sdk/cosmosdb/cosmos/src/queryIterator.ts

```diff
@@ -368,6 +419,14 @@ export class QueryIterator<T> {
 
   private initPromise: Promise<void>;
   private async init(diagnosticNode: DiagnosticNodeInternal): Promise<void> {
+    // add rid to options if encryption is enable for client
+    if (this.container && this.clientContext.enableEncryption) {
```

Comment:
it should be part of container creation, not part of query iterator initialization